lib/node_modules/@stdlib/blas/ext/last-index-of/lib/assign.js

```diff
@@ -0,0 +1,223 @@
+*
+* @param {ndarrayLike} x - input ndarray
+* @param {(ndarrayLike|*)} searchElement - search element
+* @param {(ndarrayLike|integer)} [fromIndex] - index from which to begin searching
+* @param {ndarrayLike} out - output ndarray
+* @param {Options} [options] - function options
+* @param {IntegerArray} [options.dim=-1] - dimension over which to perform operation
+* @throws {TypeError} function must be provided at least three arguments
+* @throws {TypeError} first argument must be an ndarray-like object
+* @throws {TypeError} third argument must be either an ndarray-like object or an integer
+* @throws {TypeError} output argument must be an ndarray-like object
+* @throws {TypeError} options argument must be an object
+* @throws {RangeError} dimension index must not exceed input ndarray bounds
+* @throws {RangeError} first argument must have at least one dimension
+* @throws {Error} must provide valid options
+* @returns {ndarray} output ndarray
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var zeros = require( '@stdlib/ndarray/zeros' );
+* var ndarray2array = require( '@stdlib/ndarray/to-array' );
+* var ndarray = require( '@stdlib/ndarray/ctor' );
+*
+* // Create data buffers:
+* var xbuf = new Float64Array( [ 1.0, 2.0, -3.0, 2.0, -5.0, 6.0 ] );
+*
+* // Define the shape of the input array:
+* var shape = [ 2, 3 ];
+*
+* // Define the array strides:
+* var strides = [ 3, 1 ];
+*
+* // Define the index offset:
+* var offset = 0;
+*
+* // Create an input ndarray:
+* var x = new ndarray( 'float64', xbuf, shape, strides, offset, 'row-major' );
+*
+* // Create an output ndarray:
+* var y = zeros( [ 2 ], {
+*     'dtype': 'int32'
+* });
+*
+* // Perform operation:
+* var out = assign( x, 2.0, y );
+* // returns <ndarray>
+*
+* var bool = ( out === y );
+* // returns true
+*
+* var arr = ndarray2array( out );
+* // returns [ 1, 0 ]
+*/
+function assign( x, searchElement, fromIndex, out ) {
+	var hasOptions;
+	var options;
+	var nargs;
+	var opts;
+	var fidx;
+	var iflg;
+	var ord;
+	var dt;
+	var sh;
+	var v;
+	var o;
+
+	nargs = arguments.length;
+	if ( !isndarrayLike( x ) ) {
+		throw new TypeError( format( 'invalid argument. The first argument must be an ndarray. Value: `%s`.', x ) );
+	}
+	if ( nargs < 2 ) {
+		throw new TypeError( format( 'invalid argument. Second argument must be either an ndarray or a scalar value. Value: `%s`.', searchElement ) );
+	}
+	if ( nargs < 3 ) {
+		throw new TypeError( format( 'invalid argument. Third argument must be an ndarray. Value: `%s`.', fromIndex ) );
+	}
+	// Resolve input ndarray meta data:
+	dt = getDType( x );
+	ord = getOrder( x );
+
+	// Initialize an options object:
+	opts = {
+		'dims': [ -1 ] // default behavior is to perform a reduction over the last dimension
+	};
+
+	// Initialize the `fromIndex` to the first element along a dimension:
+	fidx = 0;
```

Comment:
I think we've initialized the `fromIndex` incorrectly here, as this doesn't follow https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/lastIndexOf semantics. There, `fromIndex` specifies the first element at which to begin searching backward. Here, `fromIndex` is used similar to `indexOf` and something like a means for reducing the sub-array.

This also needs to be addressed in `blas/ext/base/*last-index-of`, as, looking at examples, those all don't appear to be parameterized correctly based upon conventional semantics.